### DISCUSSION TITLE
interface white list

### DISCUSSION THREAD

**Discussion by piniSolomon** [eProsima/Fast-DDS#4046] [Support]
hi guys,
I'm getting familiar with the fast dds project I want to make use of the interfaceWhiteList element in the transport descriptor. Once I set values in the xml element the reader and writer do no match. The values im entering are the ip addresses of both pc the network.
I also tried to put only one IP for each pc and that didn't work also.
**will be glad for some help with that**
**********************************
both pc's running ubuntu 20.04
the second pc is a VM using virtualbox

```
  <transport_descriptor>
                <transport_id>TransUDP</transport_id>
                <type>UDPv4</type>
                <sendBufferSize>9216</sendBufferSize>
                <receiveBufferSize>9216</receiveBufferSize>
                <maxMessageSize>1500</maxMessageSize>
                <non_blocking_send>true</non_blocking_send>
                <listening_ports>
                    <port>7410</port>
                </listening_ports>
                <interfaceWhiteList>
                    <address>127.0.0.1</address>
                    <address>10.1.10.1</address>
                    <address>10.1.20.1</address>
                    <address>10.1.20.102</address> <!-- the second pc-->
                    <address>10.1.10.102</address>
                </interfaceWhiteList>
            </transport_descriptor>
```

**Comment by JesusPoderoso** [comment]
Hi @piniSolomon, thank you for using Fast DDS. You have to configure the transport descriptor in each of the pc's DDS applications. Let's call them pc A and pc B. Include the pc A's IP addresses (I guess that localhost would not be needed) in the pc A's participant transport descriptors (as explained in [this section of the documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport)), as you are configuring which of the available interfaces you want the participant A to communicate through. In pc B, you should create another XML file including the pc B's IP addresses in the B participant transport descriptors.

Also looking at your configuration, the listening ports configuration will be ignored because it is only taken into account in TCP transports.

**Reply by piniSolomon** [reply]
thank you for the response I will try it

**Comment by EduPonz** [comment]
Hi @piniSolomon,

There are several not necessary things in your transport descriptor. To achieve what you want you'd only need `transport_id`, `type`, and `interfaceWhitelist`. Another important point is that the addresses you need to specify in the whitelist are those that you allow Fast DDS to use. In that sense, you should only write local addresses there, not the ones from the other PC. For example, if host A has an IP 10.1.10.101, and host B has an IP 10.1.10.102, the the participant running in host A should use a transport descriptor which interface whitelist contains 10.1.10.101, and the participant in B should use a transport descriptor which interface whitelist contains 10.1.10.102.

In addition, for the communication to work, both participants should be connected to at least one common LAN through which multicast works correctly (if it does not then you'd need to configure some other things), have you verified this? Sometimes the network configuration for virtual networks used between host and VM needs to explicitly enable multicast, and sometimes the out-of-the-box may have some firewall rules.

**Reply by piniSolomon** [reply]
thank you for the response. I rather not use multicast.

**Reply by EduPonz** [reply]
I'm leaving here some documentation on disabling multicast: https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/wifi/disable_multicast.html
